content/post/setup-kubernetes-raspberry-pi-cluster.md

```diff
@@ -70,106 +70,121 @@ The installation requries root privileges. Retrieve them by
+```console
+$ kubectl get nodes
 ```
 
 ![k8S](/images/kubernetes-setup-cluster/get-nodes.png)
 
-Finally, we need to setup flannel as the network driver. Meanwhile, [Weave](https://www.weave.works/products/weave-net/) is also available, but here we just present flannel:
-```
-curl -sSL https://rawgit.com/coreos/flannel/master/Documentation/kube-flannel.yml | sed "s/amd64/arm/g" | kubectl create -f -
+Finally, **we need to setup the Pod network driver** from the master node. 
+We picked flannel here because that's the only available solution for ARM at the moment. That's subject to change in the future though.
+
+```console
+$ curl -sSL https://rawgit.com/coreos/flannel/master/Documentation/kube-flannel.yml | sed "s/amd64/arm/g" | kubectl create -f -
 ```
 
 ![k8S](/images/kubernetes-setup-cluster/flannel.png)
 
-
 Start a small service
 -------------------
 Let's just run a simple service so see if the cluster actually can publish a service! Run:
 
+```console
+$ kubectl run hypriot --image=hypriot/rpi-busybox-httpd --replicas=3 --port=80
 ```
-kubectl run hypriot --image=hypriot/rpi-busybox-httpd --replicas=3 --port=80
-```
+
 This command starts the set of containers from the image **hypriot/rpi-busybox-httpd** and defines the port the container listens on at **80**. The service will be **replicated with 3 containers**.
 
 Next, publish the containers to be accessible outside the cluster:
 
-```
-kubectl expose deployment hypriot --port 80 --type NodePort
+```console
+$ kubectl expose deployment hypriot --port 80 --type NodePort
 ```
 
-In contrast to Docker, K8s itself does not provide an option to define a specific port that listens for a service. According to Lucas is this a design decision, since routing of incoming requests to the front server of the cluster should be handled by a loadbalancer or a webserver. Also see [ingress](http://kubernetes.io/docs/user-guide/ingress/), an extension of K8s that can be configured to route incoming requests to the correct target.
+In contrast to Docker Swarm, Kubernetes itself does not provide an option to define a _specific port_ that listens for a service on the master node. According to Lucas is this an important design decision; routing of incoming requests should be handled by a third party loadbalancer or a webserver, not the core product. The core Kubernetes should be lean and extensible, and encourage others to build tools on top of it for their specific needs.
 
-K8s assigns ports in the range of 30000-32767 to a deployment (think of 'deployment' as a 'Docker service'). To get the port of the service that we just created, run:
+Often Kubernetes offers one possible implementation for a given task, like `kubeadm`. Kubernetes can be set up on so many other custom ways, but since most of the users want it simple, it makes sense to build a opinionated tool that has sane defaults and works for the maximum amount of people.
```

Comment:
This paragraph and the next one should IMHO not be part of the how to, because it interrupts the  straight-forward instructions. 
Since this info might still be interesting for at least a minority of readers, let's change these paragraphs into a quotation, as it is here: http://blog.hypriot.com/post/high-availability-with-docker/#results, and start it with a "side note about routing in Kubernetes" or similar.